splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -1140,6 +1152,28 @@ public ExecRow next() {
         }
     }
 
+    private static long getNumOfPartitions(TableDescriptor td) throws StandardException {
+        String tableId = Long.toString(td.getBaseConglomerateDescriptor().getConglomerateNumber());
+        byte[] table = Bytes.toBytes(tableId);
+        Partition root = null;
+        try {
+            long result = 1;
+            root = SIDriver.driver().getTableFactory().getTable(table);
+            if (root != null)
+                result = root.subPartitions(true).size();
+            return result;
+        } catch (Exception ioe) {
+            throw StandardException.plainWrapException(ioe);
+        } finally {
+            try {
+                if (root != null)
+                    root.close();
+            } catch (IOException e) {
+                throw StandardException.plainWrapException(e);
+            }
+        }
+    }
```

Comment:
I think you could use here a try-with-resources here to simplify the code a bit:
```
try (Partition root = SIDriver.driver().getTableFactory().getTable(table)) {
        return root != null ?  root.subPartitions(true).size() : 1;
} catch (Exception ioe) {
        throw StandardException.plainWrapException(ioe);
}
```
If I'm not mistaken the catch block will also catch an exception on root.close()

Reply:
Thanks @dgomezferro ! I've made the simplification as suggested.